staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -31,6 +31,17 @@ type DelayingInterface interface {
 	Interface
 	// AddAfter adds an item to the workqueue after the indicated duration has passed
 	AddAfter(item interface{}, duration time.Duration)
+	// AddWithOptions performs an add with additional client selected options.
+	AddWithOptions(item interface{}, opts DelayedAddOptions)
+	// AddAfterWithOptions performs a delayed add with additional client selected options.
+	AddAfterWithOptions(item interface{}, duration time.Duration, opts DelayedAddOptions)
+}
+
+// DelayedAddOptions defines a set of additional options that a caller may add to an AddWithOptions() or
+// AddAfterWithOptions() call to the delaying queue.
+type DelayedAddOptions struct {
+	DropIfWaiting bool
+	AsyncAdd      bool
```

Comment:
these need godoc to explain what they do (especially DropIfWaiting, to clarify whether the thing dropped is the thing waiting or the thing being added)

Reply:
Have done